scripts/esm-post-process.js

```diff
@@ -0,0 +1,20 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+const glob = require("glob");
+const replaceStream = require("replacestream");
+
+glob("lib/**/*.js", (_, filePaths) => {
+  for (const filePath of filePaths) {
+    const parts = filePath.split(".")
+    parts.pop();
+    const mjsFileName = parts.join(".") + ".mjs"
+    fs.createReadStream(path.join(__dirname, "..", filePath))
+      .pipe(replaceStream(/from '(\.?\.\/[^']*)'/g, "from '$1.mjs'"))
+      .pipe(fs.createWriteStream(path.join(__dirname, "..", mjsFileName)))
+      .on("close", () => {
+        fs.unlinkSync(filePath);
+      });
+  }
+});
```

Comment:
Wondering if we can pimp our Rollup or TS build configurations instead of doing custom post-processing?

Reply:
I dug a bit, but it seems like this is not possible with typescript at the moment: https://github.com/microsoft/TypeScript/issues/18442

Someone posted a rollup solution over here: https://github.com/microsoft/TypeScript/issues/18442#issuecomment-749896695

However, afaik rollup is rather used for building a single file instead of transpiling many files? But maybe I am wrong?

The script seemed the easiest solution with the fewest additional dependencies to me (that also works on all os platforms).

Reply:
Yeah, TS is still struggling and, you're right, Rollup is for bundling to a single file.

The script is nice, very straight forward. Was just a thought.

Reply:
My only concern would be too many open files at the same time, however, I guess given the number of files we got right now that shouldn't be too bad. It can always be optimized later.

Reply:
Thats fine for now. Premature optimization is the root of all evil haha.

We can get correct ES modules with `.mjs` imports using this Rollup config:
```js
import typescript from '@rollup/plugin-typescript';
export default {
  preserveModules: true,
  input: './src/index.ts',
  output: [
    {
      dir: 'lib',
      format: 'es',
      entryFileNames: '[name].mjs',
    },
  ],
  plugins: [typescript()],
};
```

**However,** [Rollup seems to "force" the use of `tslib`](https://github.com/rollup/plugins/issues/265#issuecomment-599339539). Because of this, it bundles a `node_modules` folder with `tslib` which I dont like at all. Firstly because we need the `__asyncValues` helper function [exclusively at one place](https://github.com/enisdenjo/graphql-ws/blob/9daef127ad1a1205abba9f1e896921492552d39b/src/server.ts#L675), and secondly, it looks ugly.